pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -135,11 +135,6 @@ def advance(self, dataloader_iter: Iterator, **kwargs: Any) -> None:
             # ------------------------------------
             # TRAINING_STEP + TRAINING_STEP_END
             # ------------------------------------
```

Comment:
I think this can be simplified. What do you think?

```python
    def advance(self, dataloader_iter: Iterator, **kwargs: Any) -> None:
        """Runs a single training batch.

        Args:
            dataloader_iter: the iterator over the dataloader producing the new batch

        Raises:
            StopIteration: When the epoch is canceled by the user returning -1
        """
        takes_iterator = isinstance(self.batch_loop, IteratorBatchProcessor)
        if takes_iterator:
            batch_input = (dataloader_iter, )
        else:
            _, (batch, is_last) = next(dataloader_iter)
            batch_input = (batch, self.batch_idx)

        self.batch_progress.increment_ready()

        # ------------------------------------
        # TRAINING_STEP + TRAINING_STEP_END
        # ------------------------------------
        with self.trainer.profiler.profile("run_training_batch"):
            batch_output = self.batch_loop.run(*batch_input)

        self.batch_progress.increment_processed()

        # By contract, when taking `dataloader_iter` as an argument, `training_step` is responsible for reporting
        # `is_last` in the result dict, which is used to determine the stop condition for the epoch. So as long as
        # `advance` is invoked, it's correct to assume that there are more batches to be processed.
        self.is_last_batch = is_last if not takes_iterator else batch_output.is_last
```

Reply:
This would be entirely removed in next PR.